code/modules/power/cells/power_cells.dm

```diff
@@ -132,3 +132,16 @@
 	overlays.Cut()
 	target.nutrition += amount
 	user.custom_emote(message = "connects \the [src] to [user == target ? "their" : "[target]'s"] charging port, expending it.")
+
+/obj/item/weapon/cell/emergency_light
+	name = "miniature power cell"
+	desc = "A tiny power cell with a very low power capacity. Used in light fixtures to power them in the event of an outage."
+	maxcharge = 120 //Emergency lights use 0.2 W per tick, meaning ~10 minutes of emergency power from a cell
+	matter = list("glass" = 20)
+	w_class = ITEMSIZE_TINY
+
+/obj/item/weapon/cell/emergency_light/Initialize()
+	. = ..()
+	var/area/A = get_area(src)
+	if(!A.lightswitch || !A.light_power)
```

Comment:
So only drain the initial charge if it neither has a light switch nor initial power? i.e., if it has a light switch but no power, or if it has power but no switch (or both, implicitly), give it cell charge?

Reply:
Not quite.  It's drain the initial charge if either the lights are switched off in that area or there's no power in the area's lighting power channel.  So give it charge only if the lights are on and powered.